src/functions/updateTechRecordStatus.ts

```diff
@@ -0,0 +1,28 @@
+import TechRecordsDAO from "../models/TechRecordsDAO";
+import TechRecordsService from "../services/TechRecordsService";
+import HTTPResponse from "../models/HTTPResponse";
+import {APIGatewayEvent} from "aws-lambda";
+import {STATUS, HTTPRESPONSE} from "../assets/Enums";
+import S3BucketService from "../services/S3BucketService";
+import { S3 } from "aws-sdk";
+
+export async function updateTechRecordStatus(event: APIGatewayEvent) {
+    const s3BucketService = new S3BucketService(new S3());
+    const techRecordsService = new TechRecordsService(new TechRecordsDAO(), s3BucketService);
+
+    const vin = event.pathParameters!.vin;
+    const testStatus: string = event.queryStringParameters!.testStatus;
+    const testResult: string = event.queryStringParameters!.testResult;
+    const testTypeId: string = event.queryStringParameters!.testTypeId;
+    const newStatus = event.queryStringParameters ? STATUS[event.queryStringParameters.newStatus as keyof typeof STATUS] : undefined;
```

Comment:
Seems... hacky. Precisely what I used string literal type to avoid. Does this offer run time checking?

Reply:
I didn't test the runtime checking (but it should check it). I used it this way to be able to use a parameter of type `STATUS` in `updateTechRecordStatusCode`